pkg/streamingpromql/operators/aggregations/topkbottomk/range_query.go

```diff
@@ -0,0 +1,497 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```

Comment:
Some of the comments on instant_query apply here too. I haven't duplicated them.

Reply:
I think I've addressed all of them, let me know if I've missed something.